cmake/modules/west.cmake

```diff
@@ -106,3 +106,9 @@ else()
     set(WEST WEST-NOTFOUND CACHE INTERNAL "West")
   endif()
 endif()
+
+if(DEFINED WEST_PYTHON AND WIN32)
+  # Replace windows back-slashes in path with forward-slashes to prevent issues
+  # with the back-slashes being wrongly interpreted as escape sequence codes.
+  file(TO_CMAKE_PATH "${WEST_PYTHON}" WEST_PYTHON)
```

Comment:
`WEST_PYTHON` path is already adjusted here:
https://github.com/nrfconnect/sdk-zephyr/blob/c2859f736bfc959b91cca472ba32563afccea0bc/cmake/modules/west.cmake#L15-L18

so we should move this adjustment upwards and let CMake take care of handling the OS, that is if on Linux and path already contains `/`, then CMake will keep those.

And while doing so, feel free to adjust `get_filename_component(... REAL_PATH)` to the never `file(REAL_PATH ...)`.
That can be done because Zephyr now requires CMake >=3.20.